src/targets/gpu/include/migraphx/gpu/context.hpp

```diff
@@ -272,8 +274,40 @@ struct context
 
         auto v_streams        = v.at("streams");
         std::size_t n_streams = v_streams.without_key().to<std::size_t>();
+        this->current_device  = std::make_shared<hip_device>(0, n_streams);
+#ifdef MIGRAPHX_HAS_FIND_2_API
+        {
+            auto v_gfx_arch = v.at("gfx_arch").to<std::string>();
+
+            auto v_cu_count        = v.at("cu_count");
+            std::size_t n_cu_count = v_cu_count.without_key().to<std::size_t>();
 
-        this->current_device = std::make_shared<hip_device>(0, n_streams);
+            std::string v_miopen_version = v.at("miopen_version").to<std::string>();
+
+            std::string current_gfx_arch       = this->current_device->get_device_name();
+            std::size_t current_cu_count       = this->current_device->get_cu_count();
+            std::string current_miopen_version = get_miopen_version();
+            if(n_cu_count != current_cu_count or v_gfx_arch != current_gfx_arch)
```

Comment:
I dont think the difference in performance for different CUs is big enough that we should spam the user. It would be better to have warnings when we use untuned configs from MIOpen. This would be a lot more important because there could be a 10x gap from this.

Reply:
> I dont think the difference in performance for different CUs is big enough that we should spam the user. 

Users would only see warnings when they deserialize the model first. So, warnings would only appear minimally and only when there is setup/configuration mismatch.

If saving CUs information results in perf gain(minimal however),  IMO, it is better to save CU info. And on most hardware configurations, CUs wouldn't change. So i don't expect to see cases of warnings appearing because of CUs mismatch.

Reply:
> If saving CUs information results in perf gain(minimal however), IMO, it is better to save CU info.

Then we should have an env var to enable these messages, as most of the time it wont make a difference.